pkgs/development/python-modules/torch/default.nix

```diff
@@ -1,6 +1,6 @@
 { stdenv, lib, fetchFromGitHub, fetchpatch, buildPythonPackage, python,
   cudaSupport ? false, cudaPackages, magma,
-  mklDnnSupport ? true, useSystemNccl ? true,
+  mklDnnSupport ? stdenv.isLinux, useSystemNccl ? true,
```

Comment:
Hmm, this poses the question of whether we should touch `x86_64-darwin`, if we were only looking at `aarch64-darwin` so far.

1. Does `x86_64-darwin` lose performance too, with mkldnn on?
2. If there's an Apple-specific library that provides the same functionality as mkldnn and that pytorch ends up consuming in your build, we should probably always use it.


I'm looking at `'wheel-py3_10-cpu-build/10_Build PyTorch binary.txt'` from one of the x86 darwin actions upstream, they seem to have these flags set:

```
--   USE_MKL               : ON
--   USE_MKLDNN            : ON
--   USE_MKLDNN_ACL        : OFF
--   USE_MKLDNN_CBLAS      : OFF
```

Ultimately, if we can manually check that, having the rest of your change merged, `mklDnnSupport = false` doesn't harm performance on `x86_64-darwin`, I think we're good to keep `isLinux`

Reply:
Also, let's add a comment referring to this PR and explaining that MKLDNN negatively impacts performance on M1

Reply:
Yes, please add a comment otherwise this gets forgotten again

Reply:
> Also, let's add a comment referring to this PR and explaining that MKLDNN negatively impacts performance on M1

Done in the new push.